Mathlib/Data/Set/Lattice.lean

```diff
@@ -686,6 +686,12 @@ theorem iInter_iInter_eq_or_left {b : β} {p : β → Prop} {s : ∀ x : β, x =
     ⋂ (x) (h), s x h = s b (Or.inl rfl) ∩ ⋂ (x) (h : p x), s x (Or.inr h) := by
   simp only [iInter_or, iInter_inter_distrib, iInter_iInter_eq_left]
 
+lemma iUnion_sum {s : α ⊕ β → Set γ} : ⋃ x, s x = (⋃ x, s (.inl x)) ∪ ⋃ x, s (.inr x) := by
+  ext; simp only [mem_iUnion, Sum.exists, mem_union]
```

Comment:
Didn't test but this should work, same below with `iInf_sum`. I think that set lemmas that are special cases of complete lattice lemmas should reuse the proofs.
```suggestion
lemma iUnion_sum {s : α ⊕ β → Set γ} : ⋃ x, s x = (⋃ x, s (.inl x)) ∪ ⋃ x, s (.inr x) :=
  iSup_sum
```

Reply:
That makes sense, thank you!